lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -297,6 +368,15 @@ sub collect_individual_alignments {
             };
         }
 
+        my $test_name;
+        if ($ENV{GENOME_SOFTWARE_RESULT_TEST_NAME}) {
+            $test_name = $ENV{GENOME_SOFTWARE_RESULT_TEST_NAME};
+        } elsif ($self->test_name) {
```

Comment:
I think Feiyu and I both agree with you. We had the same thought when we first reviewed the existing code, and only included this section in case we were wrong.